crowbar_framework/config/initializers/checks.rb

```diff
@@ -0,0 +1,47 @@
+#
+# Copyright 2015, SUSE LINUX GmbH
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+unless ENV["SKIP_CHECKS"] && ENV["SKIP_CHECKS"] == "yes"
+  network_checks = Crowbar::Checks::Network.new
+
+  unless network_checks.fqdn_detected?
+    raise("Unable to detect fully-qualified hostname.")
```

Comment:
I like the split with seperate checks and raise calls, but what about raising exceptions directly within the check methods?